Utils.h

```diff
@@ -15,35 +15,17 @@ inline std::string basename(const std::string& pathname) {
             pathname.end()};
 }
 
-inline std::string intToScalaNat(int v, std::string accumulator = ""){
-
-    if(v > 0){
-        int last_digit = v % 10;
-        int rest = v / 10;
-
-        if(accumulator == ""){
-            return intToScalaNat(rest, "_" + std::to_string(last_digit));
-        } else{
-            return intToScalaNat(rest, "Digit[_" + std::to_string(last_digit) + ", " + accumulator + "]");
-        }
-    } else{
+inline std::string uint64ToScalaNat(uint64_t v, std::string accumulator = "") {
+    if (v == 0)
         return accumulator;
-    }
-}
-
-inline std::string uint64ToScalaNat(uint64_t v, std::string accumulator = ""){
 
-    if(v > 0){
-        int last_digit = v % 10;
-        int rest = v / 10;
+    int last_digit = v % 10;
+    int rest = v / 10;
```

Comment:
CLion gives warnings about type casts. Is it fine to change these lines to following: ?
```c++
int last_digit = static_cast<int>(v % 10);
uint64_t rest = v / 10;
```

Reply:
How about using `auto`? That should keep everything unsigned.

Reply:
Also, maybe we should configure some compiler warning flags and fail the build.